Add download/upload feature for domain pattern block list

MrPetovan opened the pull request:
Closes #11775

Quix0r reviewed (approved):
Looks fine to me.

tobiasd commented:
I would prefer to have the two exported lists of blocked servers in the same format. The console command is using CSV, here you use JSON if I see correctly. Personally I prefer CSV as it is more human readable, but as this is mostly meant as system-interchange format and not human readable, I would not mind if the console tool was switched to JSON as well.

On the other hand, contact list exports are done as CSV file (as do other platforms in the Fediverse) so maybe stick with CSV as export format in all cases.

MrPetovan commented:
I wasn't aware there was a console command to export the blocklist, I'll switch the UI pages to CSV.

nupplaphil commented on src/Module/Admin/Blocklist/Server/Import.php:
```diff
@@ -0,0 +1,117 @@
+namespace Friendica\Module\Admin\Blocklist\Server;
+
+use Friendica\Core\Renderer;
+use Friendica\DI;
```
Please replace all `DI::*` with a proper constructor initialization:)

MrPetovan replied:
I forgot it works with Modules now!

MrPetovan committed:
Move system messages from boot to own class

MrPetovan committed:
Add download feature for domain block list

- Add new /blocklist/domain/download route
- Add link on /friendica page

MrPetovan committed:
Add upload feature for domain pattern block list

- Add new /admin/blocklist/server/import route
- Add form on domain pattern blocklist index page

MrPetovan committed:
Updated main translation file after adding strings
